### PR TITLE
Various WGSL fixes.

### LINKED ISSUE #5227: slang-test/WGPU: unfilterable-float type sampler used to filter
This issue was split from https://github.com/shader-slang/slang/issues/5173.

Affected tests under tests/compute (currently excluded):

1. `func-param-legalize`

Output from slang-test, for `func-param-legalize`:

    ...
    WGPU error: Texture binding (group:0, binding:0) is TextureSampleType::UnfilterableFloat but used statically with a sampler (group:0, binding:1) that's SamplerBindingType::Filtering
     - While validating compute stage ([ShaderModule (unlabeled)], entryPoint: computeMain).
     - While calling [Device].CreateComputePipeline([ComputePipelineDescriptor]).

    WGPU error: [Invalid ComputePipeline (unlabeled)] is invalid.
     - While encoding [ComputePassEncoder (unlabeled)].SetPipeline([Invalid ComputePipeline (unlabeled)]).
     - While finishing [CommandEncoder (unlabeled)].

    WGPU error: [Invalid CommandBuffer] is invalid.
     - While calling [Queue].Submit([[Invalid CommandBuffer]])
    ...

Generated WGSL for `slangc -o test.wgsl -target wgsl -stage compute -entry computeMain %SLANG_SRC%\tests\compute\func-param-legalize.slang`:

    @binding(0) @group(0) var diffuseMap_0 : texture_2d<f32>;
    
    @binding(1) @group(0) var samplerState_0 : sampler;
    
    @binding(2) @group(0) var<storage, read_write> outputBuffer_0 : array<f32>;
    
    struct Param_0
    {
         base_0 : f32,
    };
    
    fn run_0( p_0 : Param_0,  p_tex_0 : texture_2d<f32>,  p_samplerState_0 : sampler) -> vec4<f32>
    {
        return (textureSampleLevel((p_tex_0), (p_samplerState_0), (vec2<f32>(0.0f)), (0.0f))) + p_0.base_0;
    }
    
    @compute
    @workgroup_size(1, 1, 1)
    fn computeMain(@builtin(global_invocation_id) dispatchThreadID_0 : vec3<u32>)
    {
        var p_1 : Param_0;
        p_1.base_0 = -0.5f;
        var _S1 : vec4<f32> = run_0(p_1, diffuseMap_0, samplerState_0);
        outputBuffer_0[i32(0)] = _S1.x;
        outputBuffer_0[i32(1)] = _S1.y;
        outputBuffer_0[i32(2)] = _S1.z;
        outputBuffer_0[i32(3)] = _S1.w;
        return;
    }

### LINKED ISSUE #5281: WGSL: switch statement without default clause
Affected tests:
- tests/bugs/nested-switch.slang

    WGPU error: Error while parsing WGSL: :9:13 error: switch statement must have a default clause
                switch(r_0)
                ^^^^^^

### LINKED ISSUE #5286: WGSL: Emitting 4079520790 as 'i32'
Affected tests:
- tests/ir/string-literal.slang

### LINKED ISSUE #5287: WGSL: texturesamplelevel offset must be const
Affected tests:
- tests/language-feature/constants/constexpr-loop.slang

See https://www.w3.org/TR/WGSL/#texturesamplelevel

### LINKED ISSUE #5289: WGSL: Ternary operator emitted
Affected tests:

- tests/autodiff-dstdlib/dstdlib-abs.slang

WGPU errors:

    WGPU error: Error while parsing WGSL: :11:47 error: invalid character found
        var _S1 : f32 = ((*dpx_0).primal_0 > 0.0f ? 1.0f : (*dpx_0).primal_0 == 0.0f ? 0.0f : -1.0f) * dOut_0;

### LINKED ISSUE #5290: slang-test/WGPU: autodiff tests failing due to mismatching values
Affected tests:
- autodiff/matrix-arithmetic-fwd.slang
- tests/autodiff/reverse-loop-checkpoint-test.slang

It seems some values are mismatching! For example from `autodiff/matrix-arithmetic-fwd`:

    EXPECTED{{{
    type: float
    8.0
    83.0
    0.0
    0.0}}}
    ACTUAL{{{
    type: float
    24.000000
    83.000000
    0.000000
    0.000000
    }}}

### PR DESCRIPTION
**Pull request by csyonghe**
Closes #5281.
Closes #5286.
Closes #5287.
Closes #5290.
Closes #5227.
Closes #5289.

### CODE REVIEW

File: tests/bugs/nested-switch.slang [hunk 1]
```diff
@@ -3,7 +3,7 @@
 //TEST(compute):COMPARE_COMPUTE: -shaderobj
 //TEST(compute):COMPARE_COMPUTE:-vk -shaderobj
 //TEST(compute):COMPARE_COMPUTE:-cpu -shaderobj
-//DISABLE_TEST(compute):COMPARE_COMPUTE_EX:-wgpu
+//TEST(compute):COMPARE_COMPUTE:-wgpu
```
Comment (jkwak-work) [inline tests/bugs/nested-switch.slang:6]: I am not sure it does, but shouldn't this have `-shaderobj` like other lines?
```
//TEST(compute):COMPARE_COMPUTE:-wgpu -shaderobj
```

**Review by jkwak-work — APPROVED** [review]
Looks good to me.

File: tests/bugs/nested-switch.slang (hunk 1 above)
Comment (csyonghe) [reply tests/bugs/nested-switch.slang:6]: That flag is no longer meaningful. Every test is `-shaderobj` now, and this flag should be removed from all tests.

**Pull request merged**